app/src/androidTest/java/com/github/se/project/ui/lesson/RequestedLessonsTest.kt

```diff
@@ -84,6 +106,16 @@ class LessonsRequestedScreenTest {
               longitude = 0.0))
   private val requestedLessonsFlow = MutableStateFlow(mockLessons)
 
+  // Mock NetworkStatusViewModel to control the network status state
+  private val mockIsConnected = MutableStateFlow(true)
+
+  private val mockNetworkStatusViewModel =
+      object :
+          NetworkStatusViewModel(
+              application = androidx.test.core.app.ApplicationProvider.getApplicationContext()) {
+        override val isConnected = mockIsConnected
+      }
+
```

Comment:
Good mock of the NetworkStatusViewModel, I think that it might be better to remove "mock" from the name as we know that it is a mock and it makes just the reading of the variables harder as all start with mock.